recursion/writing-recursive-code.md

```diff
@@ -0,0 +1,183 @@
+# Writing Recursive Code
+
+## Learning Goals
+
+- Construct a recursive solution from a word problem
+
+## Considerations for a Recursive Function
+
+When writing a recursive function, the following prompts will help construct different parts of a recursive function.
+
+- What is the base case? What is the version of the problem where we know the answer without recursing?
+- How can we describe the repeated problem that gets divided into smaller versions of the problem?
+- What information does each smaller problem need?
+- What information do we want back from each problem?
+- How do we expect examples to behave?
+
+### Determine the Base Case
+
+The base case is the the version of the problem where we can determine the answer without recursing. Generally, the base case is used as the condition which ends the recursion.
+
+We may be able to determine the base case from the given problem statement, or we may need to ask clarifying questions.
+
+### Describe the Repeated Smaller Problem
+
+What is the problem that needs to be solved? What do "smaller" versions of the problem look like? The answers to these questions will help us determine when and how to make the recursive function call.
+
+We'll use this answer to help us visualize the solution, as we'll use the remaining prompts to determine details.
+
+### Determine the Information the Smaller Problem Needs
+
+Each version of the smaller problem needs input. What pieces of input are needed?
+
+The answer to this question will help us determine the _parameters_ of our recursive function.
+
+At this moment, we should also consider how the parameter moves towards the base case. Recall that if the parameters never move towards the base case, we'll end up with infinite recursion.
+
+### Determine the Information We Want Back
+
+Given the problem statement, what is the information that each problem should give back?
+
+The answer to this question will determine the _return_ value of our function.
+
+### Create Example Tables
+
+When we have a draft of our recursive function, we can create example tables to check our work.
+
+Here is a sample table with sample headers that we could use. Each row in the table should represent a function call that is on the system stack. The "function call" column in this table is meant to identify a function call on the system stack.
```

Comment:
it's not really a full "table", but we can use it to build one
system stack → call stack
the last sentence sounds redundant. if the use for the first column is unclear, it should become clear when teh sample is shown a bit later
```suggestion
Here is a set of sample headers that we could use to build an example table. Each row in the table should represent a function call that is on the call stack.
```